opm/simulators/flow/FlowsContainer.cpp

```diff
@@ -0,0 +1,335 @@
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+  Consult the COPYING file in the top-level source directory of this
+  module for the precise wording of the license and the list of
+  copyright holders.
+*/
+#include <config.h>
+#include <opm/simulators/flow/FlowsContainer.hpp>
+
+#include <opm/input/eclipse/EclipseState/SummaryConfig/SummaryConfig.hpp>
+#include <opm/input/eclipse/Schedule/Schedule.hpp>
+#include <opm/input/eclipse/Units/UnitSystem.hpp>
+
+#include <opm/material/fluidsystems/BlackOilDefaultIndexTraits.hpp>
+#include <opm/material/fluidsystems/BlackOilFluidSystem.hpp>
+#include <opm/material/fluidsystems/GenericOilGasWaterFluidSystem.hpp>
+
+#include <opm/output/data/Solution.hpp>
+
+#include <algorithm>
+#include <tuple>
+
+namespace {
+
+    template<class Scalar>
+    using DataEntry = std::tuple<std::string,
+                                 Opm::UnitSystem::measure,
+                                 std::array<std::vector<Scalar>,6>&>;
+
+    template<int idx, class Array, class Scalar>
+    void addEntry(std::vector<DataEntry<Scalar>>& container,
+                  const std::string& name,
+                  Opm::UnitSystem::measure measure,
+                  Array& flowArray)
+    {
+        if constexpr (idx >= 0) {  // Only add if index is valid
+            container.emplace_back(name, measure, flowArray[idx]);
+        }
+    }
+
+    template<int idx, class Array, class Scalar>
+    void assignToVec(Array& array,
+                     const unsigned faceId,
+                     const unsigned globalDofIdx,
+                     const Scalar value)
+    {
+        if constexpr (idx != -1) {
+            if (!array[idx][faceId].empty()) {
+                array[idx][faceId][globalDofIdx] = value;
+            }
+        }
+    }
+
+    template<int idx, class Array, class Scalar>
+    void assignToNnc(Array& array,
+                     unsigned nncId,
+                     const Scalar value)
+    {
+        if constexpr (idx != -1) {
+            if (!array[idx].indices.empty()) {
+                array[idx].indices[nncId] = nncId;
+                array[idx].values[nncId] = value;
+            }
+        }
+    }
+
+}
+
+namespace Opm {
+
+template<class FluidSystem>
+FlowsContainer<FluidSystem>::
+FlowsContainer(const Schedule& schedule,
+               const SummaryConfig& summaryConfig)
+{
+    // Check for any BFLOW[I|J|K] summary keys
+    blockFlows_ = summaryConfig.keywords("BFLOW*").size() > 0;
+
+    // Check if FLORES/FLOWS is set in any RPTRST in the schedule
+    anyFlores_ = false;     // Used for the initialization of the sparse table
+    anyFlows_ = blockFlows_;
```

Comment:
These assignments will be overwritten by the `any_of()` calls below.  Static analysis may or may not flag that as unnecessary writes.

Reply:
indeed. inherited from original code structure.